numba/cuda/tests/cudapy/test_gufunc.py

```diff
@@ -183,6 +186,49 @@ def copy2d(A, B):
         copy2d(A, out=B)
         self.assertTrue(np.allclose(A, B))
 
+    # Test inefficient use of the GPU where the inputs are all mapped onto a
+    # single thread in a single block.
+    def test_inefficient_guvectorize(self):
+        @guvectorize(['void(float32[:], float32[:], float32[:])'],
+                     '(n),(n)->(n)', target='cuda')
+        def numba_dist_cuda(a, b, dist):
+            len = a.shape[0]
+            for i in range(len):
+                dist[i] = a[i] * b[i]
+
+        a = np.random.rand(1024 * 1024 * 32).astype('float32')
+        b = np.random.rand(1024 * 1024 * 32).astype('float32')
+        dist = np.zeros(a.shape[0]).astype('float32')
+
+        with override_config('CUDA_LOW_OCCUPANCY_WARNINGS', 1):
+            with warnings.catch_warnings(record=True) as w:
+                warnings.simplefilter('always', NumbaPerformanceWarning)
+                numba_dist_cuda(a, b, dist)
+                self.assertEqual(w[0].category, NumbaPerformanceWarning)
+                self.assertIn('Grid size', str(w[0].message))
+                self.assertIn('2 * SM count',
+                              str(w[0].message))
+
+    def test_efficient_guvectorize(self):
+        @guvectorize(['void(float32[:], float32[:], float32[:])'],
+                     '(n),(n)->(n)', nopython=True, target='cuda')
+        def numba_dist_cuda2(a, b, dist):
+            len = a.shape[0]
+            for i in range(len):
+                dist[i] = a[i] * b[i]
+
+        a = np.random.rand(1024 * 1024 * 128).astype('float32').\
+            reshape((1024 * 1024, 128))
+        b = np.random.rand(1024 * 1024 * 128).astype('float32').\
+            reshape((1024 * 1024, 128))
```

Comment:
Also to reduce the data size here please:

```suggestion
        a = np.random.rand(524288 * 2).astype('float32').reshape((524288, 2))
        b = np.random.rand(524288 * 2).astype('float32').reshape((524288, 2))
```

The leading dimension of 524288 is needed to get a big enough grid size for a large GPU, so I reduced the trailing dimension down to 2 to keep the data size as small as possible.